client/src/components/AuthForm.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+
+class AuthForm extends React.Component {
+    constructor(props){
+        super(props);
+        this.state = {
+            registering: false,
+            username: '',
+            password: '',
+            email: '',
+            firstName: '',
+            lastName: '',
+        };
+    }
+    handleChange = ev => {
+        this.setState({ [ev.target.name]: ev.target.value });
+    }
+    changeOperation = ev => {
+        ev.preventDefault();
+        this.setState((prevState) => {
+            return {
+                registering: !prevState.registering,
+                username: '',
+                password: '',
+                email: '',
+                firstName: '',
+                lastName: '',
+            };
+        })
+    }
+    handleSubmit = ev => {
+        ev.preventDefault();
+        if (this.state.registering){
+            const user = {
+                username: this.state.username,
+                password: this.state.password,
+                email: this.state.email,
+                firstName: this.state.firstName,
+                lastName: this.state.lastName,
+            };
+            this.setState({
+                username: '',
+                password: '',
+                email: '',
+                firstName: '',
+                lastName: '',
+            });
+            this.props.onRegister(user);
+        } else {
+            const user = {
+                username: this.state.username,
+                password: this.state.password,
+            };
+            this.setState({
+                username: '',
+                password: '',
+            });
+            this.props.onLogin(user);
+        }
+    }
+    render(){
+        return (
+            <section className="auth-container">
+                {this.state.registering 
+                ? (
+                    <form onSubmit={this.handleSubmit}>
+                        <input 
+                            type="text"
+                            placeholder="Username"
+                            name="username"
+                            onChange={this.handleChange}
+                            value={this.state.username}
+                        />
+                        <input 
+                            type="password"
+                            placeholder="Password"
+                            name="password"
+                            onChange={this.handleChange}
+                            value={this.state.password}
+                        />
```

Comment:
All of your inputs look nearly the same. Why not fix that up? For example, making a method in this component rather making a different component (although that'd be a more reusable approach):
```javascript
makeInput = (name, type = 'text') => {
  const capitalize = str => str[0].toUpperCase() + str.slice(1);
  const makePlaceholder = str => capitalize(
    str.replace(/[A-Z]/g, char => ' ' + char)
  );

  return (
    <input 
       type={type}
       placeholder={makePlaceholder(name)}
       name={name}
       onChange={this.handleChange}
       value={this.state[name]}
     />
  );
};
```

and the usage, for the two inputs above and the one below for example:
```javascript
<form onSubmit={this.handleSubmit}>
  {this.makeInput('username')}
  {this.makeInput('password', 'password')}
  // ...
```
See how much less repetition there would be with a helper like this? Feel free to come up with your own implementation if this one doesn't quite suit your needs!